Feat : 컨퍼런스 도메인 및 cu 구현

Kimseungin0529 opened the pull request:
### 진행 이슈
close #7 

### 무엇을 진행했나요?
 기본적인 컨퍼런스에 대한 CU 기능 구성
 날짜 검증 로직 구현

### 기타
조회 기능은 부스, 세션 등 다른 도메인도 결합되기에 따로 이슈, PR 파서 추가하겠습니다.
API 테스트는 성공 케이스만 포스트맨으로 진행했습니다.

개인적으로 Rest Docs도 도입하고 싶어서 추후 문서 추가할때 컨트롤러 테스트도 추가하겠습니다!

[3월 7일 추가]
yml 파일 충돌은 병합 전에 기존 브랜치에 맞게 변경하겠습니다. 추후 yml 파일 형식도 정하면 좋습니다.

Kimseungin0529 committed:
style : ErrorTpye 변수명 규칙 통합으로 인한 수정

Kimseungin0529 committed:
test : Conference 정적 팩토리 메서드 테스트 구현

임시 푸쉬 과정에서 충돌로 인해 잘못된 부분은 커밋 내용과 별개로  변경했습니다.

Kimseungin0529 committed:
test : Conference 정적 팩토리 메서드 테스트 구현

임시 푸쉬 과정에서 충돌로 인해 잘못된 부분은 커밋 내용과 별개로  변경했습니다.

Kimseungin0529 committed:
test/feat : Confernce 팩토리 메서드 - 컨퍼런스명 예외 케이스 구현 / Conference 관련 커스텀 예외 구현

추가된 커스텀 예외
InvalidLocationException, InvalidNameException

Kimseungin0529 committed:
test/refactor : Conference 팩토리 메서드 - 컨퍼런스 위치 예외 케이스 구현 / 팩토리 메서드 예외 클래스 변경

Kimseungin0529 committed:
feat : conference create dto 구현

wjd4204 committed:
feat: Session CRUD 기능 구현

Kimseungin0529 committed:
feat : conference update dto 구현

leeminjung1 committed:
refactor: Restructure Member table schema (#9)

* refactor: Restructure Member table schema

* refactor: Introduce MemberRepository as a base repository

* refactor: Rename adminId to assignedAdminId

* feat: add phone column

wjd4204 committed:
fix: 상속 클래스 구현 오류 수정

Commit:
fix: Resolve Hibernate mappedBy issue in MemberInterest & Ensure @SuperBuilder respects default values

Kimseungin0529 committed:
feat/test: ConferenceService CU 기초 구현 / ConferenceService register test 구현

Kimseungin0529 committed:
feat/test: ConferenceService CU 기초 구현 / ConferenceService register test 구현

Kimseungin0529 committed:
feat : ConferenceServiceImpl register method 구현 완료

Kimseungin0529 committed:
refacotr : ConferenceUpdateRequest 수정

수정 기능이기에 빈 값을 허용해야 함. 따라서 null 허용 처리

Kimseungin0529 committed:
test : conferenceService update method test 구현

update 메서드 스펙 변경 - 파라미터

wjd4204 committed:
fix: repository 메서드 수정

Kimseungin0529 committed:
style : ConferenceServiceImpl update 초기 로직 주석 표시

Kimseungin0529 committed:
ConferenceServiceImpl update method 구현

1. ConferenceUpdateResponse class -> record 변경
2. record 변경으로 test code도 변경
3. conference 필드 변경 기능 추가

wjd4204 committed:
fix: 생성 패턴 수정

wjd4204 committed:
fix: 생성 패턴 수정

Kimseungin0529 committed:
test : ConferenceServiceImpl update test 리팩토링

Kimseungin0529 committed:
refactor : ConferenceServiceImpl update 메서드 추상화 작업

Kimseungin0529 committed:
feat : ConferenceService 인터페이스 스펙 변경 및 추가

Kimseungin0529 committed:
test/chore : Conference ofExceptionName test 시나리오 구현 / application.yml 방언 추가

Kimseungin0529 committed:
feat : ConferenceController registerConference 구현

Kimseungin0529 committed:
feat : ConferenceController updateConference 구현

추가로 반환값 구체화

Kimseungin0529 committed:
refactor : ConferenceController @RequestBody, @Valid 추가

wjd4204 commented on src/main/java/com/synergy/backend/domain/conference/api/ConferenceController.java:
```diff
@@ -0,0 +1,29 @@
+        return ApiResponse.ok(conferenceService.registerConference(request), 201);
+    }
+
+    @PatchMapping("{id}")
```
"/"가 빠진것 같습니다!

Kimseungin0529 replied:
반영했습니다!

wjd4204 commented on src/main/java/com/synergy/backend/domain/conference/dto/requset/ConferenceCreateRequest.java:
```diff
@@ -0,0 +1,20 @@
+public record ConferenceCreateRequest(
+        @NotBlank(message = "컨퍼런스명은 필수입니다. 공백 이하는 불가능합니다.")
+        String name,
+        @Future(message = "시작 날짜는 미래여야 합니다.")
```
Validation에 Future이라는 기능을 통해 미래 날짜를 검증하는지 몰랐네요!
하나 알아갑니다!

wjd4204 commented on src/main/java/com/synergy/backend/domain/conference/dto/requset/ConferenceUpdateRequest.java:
```diff
@@ -0,0 +1,25 @@
+@Getter
+@NoArgsConstructor
+@AllArgsConstructor
+public class ConferenceUpdateRequest{
```
dto는 record로 통일하는 것이 좋아보입니다!
또한 Request보다는 ReqDto로 클래스의 의도를 확실히 전달하는 것은 어떨까요?

Kimseungin0529 replied:
넵 통일하겠습니다👍

wjd4204 commented on src/main/java/com/synergy/backend/domain/conference/entity/Conference.java:
```diff
@@ -49,12 +51,22 @@ private Conference(String name, TimePeriod period, String location) {
 
     public static Conference of(String name, TimePeriod period, String location) {
         if (name.length() > MAX_NAME_LENGTH || name.isBlank()) {
```
사전에 validation을 통한 유효성 검증을 추가하여 해당 조건문을 대체하는 것은 어떨까요?

Kimseungin0529 replied:
개인적으로는 객체 관점에서 보면 자신 이외는 어떤 작업이 어떻게 이뤄지는지 모르기에 따로 하는 게 좋다고 생각합니다.
컨퍼런스명 길이 제한이나 공백이면 안 된다는 건 컨퍼런스 도메인 정보이기에 필수 요소라고 느낍니다.

해당 검증이 없다면 conference의 책임이 오히려 외부 세계로 넘어갔기에 책임 분리가 적절하지 않다고 봅니다!

ㅡㅡㅡㅡㅡㅡㅡㅡㅡㅡㅡㅡㅡㅡㅡㅡㅡㅡㅡㅡㅡㅡㅡㅡㅡㅡ

잘못 이해했네요. 말씀해 주셔서 제대로 이해했습니다! 참고할게요.

wjd4204 commented on src/main/java/com/synergy/backend/domain/conference/service/ConferenceDetailResponse.java:
record로 변경해주시면 추후 수정에 용이할 것 같습니다!

Kimseungin0529 replied:
언급해 주신 부분은 수정하고 말씀드릴게요👍

wjd4204 commented on src/main/java/com/synergy/backend/domain/member/repository/MemberRepository.java:
이 repository는 뭔가요? 설명 부탁드립니다!

Kimseungin0529 replied:
저도 이제 봤네요. 이미 develop 브랜치에 있는데도 추가가 되었네요.
충돌 문제로 rebase로 정리했는데 그 과정에서 Member 도메인 부분이 추가된 거 같습니다. 제가 개발한 부분은 아닙니다.
develop 브랜치와 비교하고 민정님한테 확인받아 보겠습니다!

Kimseungin0529 committed:
refactor : 일부 형식 변경

1. ConferenceController updateConference url 오타 수정
2. ErrorType errorcode 수정
3. 등

leeminjung1 reviewed (approved):
LGTM👍 고생하셨습니다.

wjd4204 reviewed (approved):
수고하셨습니다!

Kimseungin0529 committed:
Merge branch 'develop' into feat/#7/컨퍼런스-cru-도메인

Signed-off-by: Seungjin <54787442+Kimseungin0529@users.noreply.github.com>
